packages/firebase_messaging/firebase_messaging_platform_interface/lib/src/method_channel/method_channel_messaging.dart

```diff
@@ -131,7 +130,8 @@ class MethodChannelFirebaseMessaging extends FirebaseMessagingPlatform {
   // Messaging requests. See this issue:
   // https://github.com/firebase/flutterfire/issues/10625
   Future<void> _APNSTokenCheck() async {
-    if (Platform.isMacOS || Platform.isIOS) {
+    if (defaultTargetPlatform == TargetPlatform.iOS ||
+        defaultTargetPlatform == TargetPlatform.macOS) {
```

Comment:
`defaultTargetPlatform` doesn't care if the platform is running on web browser or not, it could be `TargetPlatform.iOS` but on web browser, we need to handle this case as well